### ISSUE TITLE
Xsalsa20 with poly1305

### ISSUE THREAD

**Issue by Jirido** [DanielDe/org-web#132]
Hi. Ahh so happy to find your program.. I do use google drive for storing my org-files but I do in principe always mount the g-drive with rclone and then I mount an encryption layer ( https://rclone.org/crypt/ ) over that , as I am a fairly political figure that aren't overly trusting. This encryption is Xsalsa20 with poly1305 done with https://godoc.org/golang.org/x/crypto/nacl/secretbox 

I would love to use your app all of the time as I love org-mode but I am just not comfortable with google having plain-text access to my files. I'm no te**orist or such but just cautious and a secretive person. The world look more and more mad by the day to me.

How would you think about implementing a decryption - encryption layer ? The key could be stored locally or in a good open source password manager like KeePassXC that have good integration features.. 
I have a folder that contain the encrypted files (also names if files and folders) so maybe it could be good with a setting for what folder and if only the content or also filenames or also even folder names are encrypted.

Just a suggestion. I wish I had programming skills and just could offer a pull request.. But sadly Texts are my thing.

Any how. Lovely app you have there. It will go a long way for sure !

**Comment by DanielDe** [comment]
Hey @Jirido, I think supporting encryption/decryption in org-web is a great idea, but unfortunately it's not something I have the bandwidth to work on myself at the moment. Hopefully that changes one day, or someone else comes along to implement this themselves!

**Comment by Jirido** [comment]
Right. If i find some solution that seems to fit, I will tell. Here is some dude that may be a candidate co coder.. Or is it a gang?
https://github.com/little-core-labs/secretbox-encoding/issues/3  
I hunt an other solution also to hotlink encrypted files from google or elsewhere and get them decrypted in the client browser by js.

**Comment by Jirido** [comment]
Here it seems to be some easy and maybe suitable web solution I look in to for my web project
https://tweetnacl.js.org/#/secretbox
https://github.com/dchest/tweetnacl-js#secret-key-authenticated-encryption-secretbox
